### PR TITLE
New metric settings page - overview

### LINKED ISSUE #562: Metric Configuration Refresh
Metric Configuration is one of the most important features of Publisher; it allows agencies to declare which metrics they plan on sharing with Justice Counts. 

Based on feedback that we received from agencies and other strategic considerations, we have redesigned the interface for Metric Configuration. New designs are documented in Figma ([mockups](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?node-id=1333-5802&t=6EK5BSuoudt9uFyY-0), [interactive prototype](https://www.figma.com/proto/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?node-id=7700-21687&scaling=min-zoom&page-id=1333%3A5802&starting-point-node-id=7700%3A21687&show-proto-sidebar=1)), and are detailed below. 

Note that there might still be slight changes to this over the next few days. 

Components: 
### Overview Page
- User can switch the sector using the menu on the side. 
- Clicking into each metric takes you to the Metric-specific Page to configure the availability, frequency, and definitions (see below). 
- Metrics should be grouped according to the following: Action Required, Available, Unavailable. 
### Metric-specific Pages
- General Structure: Main Container and Side Panel 
    - The (left-hand) Side Panel consists of the metric name, metric description, links to the subpages (listed below), and an indicator for the “publishing status” of the metric. 
        - Subpages: 
        - Indicator: 
    - The Main Container contains the interactive content for the subpages, where the user will configure the metric. 
#### 1) Declare Availability: 
- The main controls for setting availability, frequency, disaggregations, etc. behave exactly the same as they did in the previous iteration. 
- Each of these questions is shortened to a short title, and a tooltip with help text will show up for each via the (i) icon (exact copy in the mocks). 
- Like the previous iteration, some prompts should only appear under certain conditions: 
    - “Starting Month” should only appear when “Annual” availability is selected. 
    - “Disaggregated by Supervision Type” should only appear if the metric is a Supervision metric. 
- For each breakdown, the user can select the disaggregations that they can report (e.g., Security Staff). (Compared to the current implementation, the user does not need to turn on the breakdown as a whole (e.g., Staff Types)). 
#### 2) Define Metrics: 
- This directly corresponds to the right-hand panel in the current implementation of Metric Configuration, which contains the “Definitions” for each metric or breakdown. 
- The page should only show the metrics and breakdowns that were marked as available in the Declare Availability page. 
- Clicking each definition opens up a modal that contains the set of “included/excluded” categories. 
    - If the contents of the modal get too long, the buttons at the bottom should stick at the bottom of the modal and other content should scroll. 

## Quick Note on Navigation
- In this iteration, the Metric Settings page is no longer a subpage of Settings; it is its own independent page.
- Ideally, we should be able to access Metric Settings directly from the navigation bar – open a dropdown for Settings, then click on Metric Settings. 
- New Navigation Bar will be detailed in a new ticket; for now, we can still access Metric Configuration from Settings.

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> New page for metric settings -> Overview

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #562 (partly)

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by hobuobi** [comment]
Looking great! Minor, non-blocking feedback, mostly for design: 
- P1: Responsiveness. (I don't think we ever finalized this!) 
- P2: Hover states for the sectors should be similar to what we do throughout the rest of the app (turns a slightly darker gray on hover) 
- P2: Seeing it in action, we probably want to have a more prominent hover state (i.e., don't just rely on the arrow). A light gray background for the row would probably do the trick. 
- P2: Slightly concerned about having to scroll to see the rest of the metrics (or maybe more concerning, all the categories).

**Comment by hobuobi** [comment]
Looking great! Minor, non-blocking feedback, mostly for design:

- P1: Responsiveness. (I don't think we ever finalized this!)
- P2: Hover states for the sectors should be similar to what we do throughout the rest of the app (turns a slightly darker gray on hover)
- P2: Seeing it in action, we probably want to have a more prominent hover state (i.e., don't just rely on the arrow). A light gray background for the row would probably do the trick.
- P2: Slightly concerned about having to scroll to see the rest of the metrics (or maybe more concerning, all the categories).

**Review by hobuobi — APPROVED** [review]

File: publisher/src/components/MetricSettings/MetricSettingsOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  const [settingsSearchParams, setSettingsSearchParams] =
+    useSettingsSearchParams();
+  const { agencyId } = useParams() as { agencyId: string };
+  const { userStore, metricConfigStore } = useStore();
+  const { initializeMetricConfigStoreValues, getMetricsBySystem } =
+    metricConfigStore;
+
+  const { system: systemSearchParam, metric: metricSearchParam } =
+    settingsSearchParams;
+  // const systemMetricKey = getActiveSystemMetricKey(settingsSearchParams);
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/MetricSettings/MetricSettingsOverview.tsx]: [12e993f](https://github.com/Recidiviz/justice-counts/pull/569/commits/12e993f431e9126030953bc14ca400c682427d2d)

File: publisher/src/components/MetricSettings/MetricSettingsOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+  if (loadingErrorMessage) {
+    return <div>Error: {loadingErrorMessage}</div>;
+  }
+
+  const currentAgency = userStore.getAgency(agencyId);
+  const showSystems =
+    currentAgency?.systems && currentAgency?.systems?.length > 1;
+  const actionRequiredMetrics = getMetricsBySystem(systemSearchParam)?.filter(
+    ({ metric }) => metric.enabled === null || metric.enabled === undefined
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/MetricSettings/MetricSettingsOverview.tsx]: [12e993f](https://github.com/Recidiviz/justice-counts/pull/569/commits/12e993f431e9126030953bc14ca400c682427d2d)

File: publisher/src/components/MetricSettings/MetricSettingsOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                    {formatSystemName(system, {
+                      allUserSystems: currentAgency?.systems,
+                    })}
+                  </Styled.SystemMenuItem>
+                );
+              })}
+        </Styled.SystemsList>
+      </Styled.OverviewWrapper>
+      <Styled.MetricsWrapper>
+        {actionRequiredMetrics && actionRequiredMetrics.length > 0 && (
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/MetricSettings/MetricSettingsOverview.tsx]: [12e993f](https://github.com/Recidiviz/justice-counts/pull/569/commits/12e993f431e9126030953bc14ca400c682427d2d)

**Review by mxosman — APPROVED** [review]
LGTM - thank you, Ivan!

**Review by lilidworkin — APPROVED** [review]
Looks great to me, thanks @ivan-kishko !

**Pull request merged**